students/stuartkershaw/lesson07/html_render.py

```diff
@@ -0,0 +1,207 @@
+#!/usr/bin/env python3
+
+
+class TextWrapper:
+    """
+    A simple wrapper that creates a class with a render method
+    for simple text
+    """
+    def __init__(self, text):
+        self.text = text
+
+    def render(self, file_out, cur_ind=""):
+        file_out.write(cur_ind)
+        file_out.write(
+            f"{self.text}"
+        )
+
+
+class Element:
+    tag_name = ''
+    indentation = 4
+    cur_tree_level = [0]
+    ignored_tag_spacing = ['', 'a', 'html']
+
+    def __init__(self, content=None, **kwargs):
+        self.content = [] if content is None else [TextWrapper(content)]
+        self.kwargs = kwargs
+
+    def append(self, content):
+        if isinstance(content, str):
+            self.content.append(TextWrapper(content))
+        else:
+            self.content.append(content)
+
+    def generate_spacing(self, tree_level):
+        self.tree_level = tree_level
+
+        return " " * self.tree_level * self.indentation
+
+    def apply_attributes(self, attrs):
+        self.attrs = attrs
+
+        for key, value in self.attrs.items():
+            self.file_out.write(f" {key}=\"{value}\"")
+
+    def render(self, file_out, cur_ind=""):
+        self.file_out = file_out
+        self.cur_ind = cur_ind
+
+        if self.tag_name not in self.ignored_tag_spacing:
+            self.cur_tree_level[0] += 1
+            self.file_out.write(
+                self.generate_spacing(self.cur_tree_level[0])
+            )
+
+        if not self.tag_name == '':
+            self.file_out.write(f"<{self.tag_name}")
+
+            if self.kwargs:
+                self.apply_attributes(self.kwargs)
+
+            self.file_out.write(f">")
+
+            if not self.tag_name == 'a':
+                self.file_out.write(f"\n")
+
+        for el in self.content:
+            el.render(self.file_out)
+
+        if self.tag_name not in self.ignored_tag_spacing:
+            self.file_out.write(
+                self.generate_spacing(self.cur_tree_level[0])
+            )
+
+        if not self.tag_name == '':
+            self.file_out.write(f"</{self.tag_name}>")
+
+        if self.tag_name not in self.ignored_tag_spacing:
+            self.cur_tree_level[0] -= 1
+            self.file_out.write("\n")
+
+
+class Html(Element):
+    tag_name = 'html'
+
+    def render(self, file_out, cur_ind=""):
+        self.file_out = file_out
+        self.cur_ind = cur_ind
+
+        self.file_out.write('<!DOCTYPE html>\n')
+
+        Element.render(self, self.file_out, self.cur_ind)
```

Comment:
I believe this is old style. You should use `super().render()`

also, `self.file_out` and `self.cur_ind` shouldn't be defined outside of `__init__`